tests/setup.py

```diff
@@ -49,12 +48,12 @@ def join_feature(f):
     # Set the columns to extract.  For env_columns, they need to match the
     # ordering of the bands in the stacked environmental image
     unwanted = ee.List([id_field, "system:index"])
-    spp_columns = spp.first().propertyNames().removeAll(unwanted).sort()
+    y_columns = y.first().propertyNames().removeAll(unwanted).sort().getInfo()
```

Comment:
Good question and maybe there is a more clever way of doing this (potentially when selecting the bands from the `X_image`)? When I do:
```
unwanted = ee.List([id_field, "system:index"])
X_columns = X.first().propertyNames().removeAll(unwanted)
```
the columns aren't sorted in the same order as the bands in the `X_image`. The band names in the GEE assets are actually ["b0", "b1", etc.], so I think I was taking the easy way out and just assigning names (e.g. `.rename(X_COLUMNS)`) rather than going to the "hard" work of correcting band names on the asset and then using `.select(X_COLUMNS)`. Then the feature collection property names could be in any order and it should work (or possibly raise an error if all names weren't found). Probably worth the latter approach?

Reply:
That seems totally fair, and I'm not sure it's worth the extra effort to avoid renaming, since this is just test code after all. I guess I was mostly thinking about this from the opposite direction: could you define `y_columns` as a constant to avoid the `getInfo` call during testing? It's probably a non-issue though since the call is only made once and is likely very fast.

Reply:
Ah, that makes sense (about `y_columns` being a constant as well).  I think I've typically shied away from this as our species matrices have been lots of columns and I was too lazy to list them all out.  (Obviously it's a cut-and-paste job if I do it once.)  I hadn't really considered the overhead of the `.getInfo` call.  I think I'll leave it for now (the whole testing suite is a bit brittle) and see if other patterns emerge if I bring in other datasets.

Reply:
Not wanting to include a giant list of columns in the code is fair! Sounds like a good plan.